source/Crew.cpp

```diff
@@ -0,0 +1,246 @@
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+*/
+
+#include "Crew.h"
+#include "Files.h"
+#include "GameData.h"
+
+using namespace std;
+
+void Crew::Load(const DataNode &node)
+{
+	if(node.Size() >= 2)
+		id = node.Token(1);
+	
+	for(const DataNode &child : node)
+	{
+		if(child.Size() >= 2)
+		{
+			if(child.Token(0) == "name")
+				name = child.Token(1);
+			else if(child.Token(0) == "parked salary")
+				parkedSalary = max((int)child.Value(1), 0);
+			else if(child.Token(0) == "place at")
+				for(int crewNumber = 1; crewNumber < child.Size(); ++crewNumber)
+					placeAt.push_back(max((int)child.Value(crewNumber), 0));
+			else if(child.Token(0) == "ship population per member")
+				shipPopulationPerMember = max((int)child.Value(1), 0);
+			else if(child.Token(0) == "salary")
+				salary = max((int)child.Value(1), 0);
+			else
+				child.PrintTrace("Skipping unrecognized attribute:");
+		}
+		else if(child.Token(0) == "avoids escorts")
+			avoidsEscorts = true;
+		else if(child.Token(0) == "avoids flagship")
+			avoidsFlagship = true;
+		else
+			child.PrintTrace("Skipping incomplete attribute:");
+	}
+}
+
+
+
+int64_t Crew::CalculateSalaries(const vector<shared_ptr<Ship>> &ships, const Ship * flagship, const bool includeExtras)
+{
+	int64_t totalSalaries = 0;
+
+	for(const shared_ptr<Ship> &ship : ships)
+	{
+		totalSalaries += Crew::SalariesForShip(
+			ship,
+			ship.get() == flagship,
+			includeExtras
+		);
+	}
+	
+	return totalSalaries;
+}
+
+
+
+const map<const string, int64_t> Crew::CrewManifest(const shared_ptr<Ship> &ship, bool isFlagship, bool includeExtras)
+{
+	int64_t crewAccountedFor = 0;
+	// A crew ID, the number on the ship, and their individual salary amount
+	tuple<string, int64_t, int64_t> cheapestCrew;
+	// Map of a crew ID to the crew type paired with the number on the ship
+	map<const string, int64_t> crewManifest;
+	
+	// Check that we have crew data before proceeding
+	if(GameData::Crews().size() < 1)
+	{
+		Files::LogError("Error: could not find any crew member definitions in the data files.");
+		return crewManifest;
+	}
+	
+	// Add up the salaries for all of the special crew members
+	for(const pair<const string, Crew> &crewPair : GameData::Crews())
+	{
+		const Crew crew = crewPair.second;
+		// Figure out how many of this type of crew are on this ship
+		int numberOnShip = Crew::NumberOnShip(
+			crew,
+			ship,
+			isFlagship,
+			includeExtras
+		);
+		
+		// Add the crew members to the manifest
+		crewManifest[crew.Id()] = numberOnShip;
+		
+		// Add the crew members to the total so far
+		crewAccountedFor += numberOnShip;
+		
+		// If this is the cheapest crew type so far, keep track of it
+		// Use non-parked salaries so that crew are consistent
+		if(get<0>(cheapestCrew).empty() || crew.Salary() < get<2>(cheapestCrew))
+			cheapestCrew = make_tuple(crew.Id(), numberOnShip, crew.Salary());
```

Comment:
The cheapest crew member for a parked ship could be different from the cheapest crew-member on a non-parked ship.

Reply:
Yes, I considered that and decided it was better for the ship to have consistent parked / non-parked crew than for it to be the cheapest possible result in each state. Hence the comment.

Reply:
Okay, clear decision.